src/integration-tests/src/main/java/com/microsoft/azure/eventhubs/checkstatus/ReceiveByDateTime.java

```diff
@@ -67,82 +71,74 @@ public static void main(String[] args)
-                    EventHubClient.DEFAULT_CONSUMER_GROUP_NAME, partitionId,
-                    EventPosition.fromEnqueuedTime(Instant.EPOCH), 2345);
-
-            System.out.println("date-time receiver created...");
-
-            try {
-                final LocalDateTime checkupStartTime = LocalDateTime.now();
-
-                // Making sure 15 minutes haven't passed since the test started
-                while (LocalDateTime.now().minusMinutes(15).isBefore(checkupStartTime)) {
-                    receiver.receive(100).thenAcceptAsync(receivedEvents -> {
-                        int batchSize = 0;
-                        if (receivedEvents != null) {
-                            for (EventData receivedEvent : receivedEvents) {
-                                System.out.print(String.format("[%s] Offset: %s, #: %s, Time: %s, PT: %s, ",
-                                        new java.util.Date(), receivedEvent.getSystemProperties().getOffset(),
-                                        receivedEvent.getSystemProperties().getSequenceNumber(),
-                                        receivedEvent.getSystemProperties().getEnqueuedTime(), partitionId));
-
-                                if (receivedEvent.getBytes() != null) {
-
-                                    try {
-                                        final String dataString = new String(receivedEvent.getBytes(), "UTF8");
-                                        final JSONObject obj = new JSONObject(dataString);
-                                        final String windowStartString = obj.getString("windowStart");
-                                        final LocalDateTime eventDateTime = LocalDateTime.parse(windowStartString,
-                                                formatter);
-                                        System.out.println("Event Time: " + windowStartString);
-
-                                        if (eventDateTime.isAfter(startTime)) {
-                                            System.out.println("Found a processed alert: " + dataString);
-                                            System.exit(0);
-                                        }
-                                    } catch (Exception e) {
-                                        System.out.println("There was a problem parsing the date of the event: "
-                                                + receivedEvent.getBytes());
-                                        continue;
-                                    }
-                                }
-                                batchSize++;
-                            }
-                        }
-
-                        System.out.println(String.format("ReceivedBatch Size: %s", batchSize));
-                    }, executorService).get();
+        Disposable subscription = consumer.receiveFromPartition(firstPartition, EventPosition.fromEnqueuedTime(Instant.EPOCH))
```

Comment:
Use `EventPosition.earliest()` instead.

Reply:
In this sample, it is designed to receive by date time.